### PR TITLE
feat(components): [el-button] support for customize loading component

### LINKED ISSUE #5439: [Feature Request] 请求更换加载按钮图标
### Existing Component
是

### Component Name
Button

### Description
希望能把加载按钮的图标替换成更加现代化的图标

### PR DESCRIPTION
**Pull request by msidolphin**
re #5439

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/examples/button/loading.vue [hunk 1]
```diff
@@ -1,9 +1,53 @@
 <template>
-  <el-button type="primary" :loading="true">Loading</el-button>
+  <el-button type="primary" loading>Loading</el-button>
+  <el-button type="primary" :loading-icon="Eleme" loading>Loading</el-button>
+  <el-button type="primary" loading>
```
Comment (btea) [inline docs/examples/button/loading.vue:4]: How to set `loading-icon` to use as a string?

Comment (msidolphin) [reply docs/examples/button/loading.vue:4]: > How to set `loading-icon` to use as a string?

if icon component already register, you can use string

Comment (btea) [reply docs/examples/button/loading.vue:4]: ok, I see

**Review by btea — APPROVED** [review]

File: docs/en-US/component/button.md [hunk 2]
```diff
@@ -97,6 +107,7 @@ button/custom
 | round             | determine whether it's a round button                       | boolean            | —                                                  | false   |
 | circle            | determine whether it's a circle button                      | boolean            | —                                                  | false   |
 | loading           | determine whether it's loading                              | boolean            | —                                                  | false   |
+| loadingIcon       | customize loading icon component                            | string / Component | —                                                  | Loading |
```
Comment (sxzz) [inline docs/en-US/component/button.md:110]:
```suggestion
| loading-icon       | customize loading icon component                            | string / Component | —                                                  | Loading |
```

File: packages/components/button/src/button.vue [hunk 3]
```diff
@@ -17,15 +17,18 @@
...
+        <component :is="loadingIcon" />
+      </el-icon>
+    </template>
     <el-icon v-else-if="icon">
       <component :is="icon" />
     </el-icon>
     <span
       v-if="$slots.default"
-      :class="[shouldAddSpace ? `${ns.em('text', 'expand')}` : '']"
+      :class="[shouldAddSpace ? ns.em('text', 'expand') : '']"
```
Comment (sxzz) [inline packages/components/button/src/button.vue:31]:
```suggestion
      :class="{[ns.em('text', 'expand')]: shouldAddSpace"
```

**Review by sxzz — APPROVED** [review]

**Review by jw-foss — APPROVED** [review]

**Review by sxzz — APPROVED** [review]

**Pull request merged**